server/console_api_explorer.go

```diff
@@ -0,0 +1,268 @@
+package server
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"github.com/gofrs/uuid"
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/heroiclabs/nakama-common/api"
+	"github.com/heroiclabs/nakama/v2/console"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"reflect"
+	"sort"
+	"strings"
+)
+
+type methodReflection struct {
+	method   reflect.Method
+	request  reflect.Type
+	response reflect.Type
+}
+
+func (s *ConsoleServer) CallRpcEndpoint(ctx context.Context, in *console.CallApiEndpointRequest) (*console.CallApiEndpointResponse, error) {
+	callCtx, err := s.extractApiCallContext(ctx, in, true)
+	if err != nil {
+		return nil, err
+	}
+
+	out, err := s.api.RpcFunc(callCtx, &api.Rpc{Id: in.Method, Payload: in.Body})
+	if err != nil {
+		return &console.CallApiEndpointResponse{
+			Body:         "",
+			ErrorMessage: err.Error(),
+		}, nil
+	}
+
+	return &console.CallApiEndpointResponse{
+		Body:         out.Payload,
+		ErrorMessage: "",
+	}, nil
+
+}
+
+func (s *ConsoleServer) CallApiEndpoint(ctx context.Context, in *console.CallApiEndpointRequest) (*console.CallApiEndpointResponse, error) {
+	r, err := reflectMethod(s.api, in.Method)
```

Comment:
I was hoping we'd have a pre-cached list of reflected methods so we can avoid doing it every single time. Is that feasible?

Also that would let us restrict which functions can be called this way. Right now a caller could try to run the `*ApiServer.Stop()` function for example - this should fail I think because of the input arguments we try to pass it but still.

Reply:
So here again, we've agreed on the channel to only cache the list. Of course we can cache the individual methods but then it's going to be better refactored into a single cache map that serves both listing and calling. The actual call will still have to use reflection every time of course.

Reply:
..also filling in the values will have to be reflected, so we could create a full cache of all methods, params and protos in something like map[<method_name>]*rpcMethodReflection which would be a struct holding a reflect type of the method, reflect of the request, response and also the reflect type of the proto message of the request.